examples/sql_bank/main.go

```diff
@@ -0,0 +1,171 @@
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/lib/pq"
+
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const (
+	maxTransfer = 999
+	numAccounts = 3
+	concurrency = 2
+	aggregate   = true
+)
+
+var (
+	dbName      = "bank"
+	tableName   = pq.QuoteIdentifier("accounts")
+	idName      = pq.QuoteIdentifier("id")
+	balanceName = pq.QuoteIdentifier("balance")
+)
+
+func moveMoney(db *sql.DB) {
+	for {
+		from, to := rand.Intn(numAccounts), rand.Intn(numAccounts)
+		if from == to {
+			continue
+		}
+		amount := rand.Intn(maxTransfer)
+		tx, err := db.Begin()
+		if err != nil {
+			log.Fatal(err)
+		}
+		var fromBalance, toBalance int
+		if err = tx.QueryRow(fmt.Sprintf("SELECT %s FROM %s WHERE id=$1", balanceName, tableName), from).Scan(&fromBalance); err != nil {
+			log.Fatal(err)
+		}
+		if err = tx.QueryRow(fmt.Sprintf("SELECT %s FROM %s WHERE id=$1", balanceName, tableName), to).Scan(&toBalance); err != nil {
+			log.Fatal(err)
+		}
+
+		// TODO: remove
+		log.Infof("amt: %3d; src: %1d, dst: %1d, srcBal: %5d, dstBal: %5d", amount, from, to, fromBalance, toBalance)
+
+		if _, err = tx.Exec(fmt.Sprintf("UPDATE %s SET %s=$2 WHERE id=$1", tableName, balanceName), from, fromBalance-amount); err != nil {
+			log.Warning(err)
+			if err = tx.Rollback(); err != nil {
+				log.Fatal(err)
+			}
+			continue
+		}
+		if _, err = tx.Exec(fmt.Sprintf("UPDATE %s SET %s=$2 WHERE id=$1", tableName, balanceName), to, toBalance+amount); err != nil {
+			log.Warning(err)
+			if err = tx.Rollback(); err != nil {
+				log.Fatal(err)
+			}
+			continue
+		}
+		if err = tx.Commit(); err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func main() {
+	db, err := sql.Open("postgres", "dbname=postgres sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", dbName)); err != nil {
+		if pqErr, ok := err.(*pq.Error); !ok || pqErr.Code.Name() != "duplicate_database" {
+			log.Fatal(err)
+		}
+	}
+
+	db, err = sql.Open("postgres", fmt.Sprintf("dbname=%s sslmode=disable", dbName))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db.SetMaxOpenConns(100)
+
+	// postgres defaults to read committed.
+	if _, err := db.Exec("SET default_transaction_isolation to serializable"); err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = db.Exec(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (%s BIGINT UNIQUE NOT NULL, %s BIGINT NOT NULL)", tableName, idName, balanceName)); err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = db.Exec(fmt.Sprintf("TRUNCATE TABLE %s", tableName)); err != nil {
+		log.Fatal(err)
+	}
+
+	for i := 0; i < numAccounts; i++ {
+		if _, err = db.Exec(fmt.Sprintf("INSERT INTO %s (%s, %s) VALUES ($1, $2)", tableName, idName, balanceName), i, 0); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	verifyBank := func() {
```

Comment:
Perhaps make this a real function. I'm not seeing a good reason for it to be a closure.